tests/integration/test_tracking.py

```diff
@@ -1200,3 +1508,48 @@ def test_memory_snapshots_tick_interval(self, tmp_path):
             _next.time - prev.time >= 20
             for prev, _next in zip(memory_snapshots, memory_snapshots[1:])
         )
+
+    def test_temporary_allocations_when_filling_vector_without_preallocating(
+        self, tmp_path
+    ):
+        # GIVEN
+        output = tmp_path / "test.bin"
+
+        # WHEN
+        with Tracker(output):
+            elements = fill_cpp_vector(2 << 10)
+
+        # THEN
+        reader = FileReader(output)
+        temporary_allocations = list(
+            reader.get_temporary_allocation_records(temporary_buffer_size=2)
+        )
+        assert elements == 512
+        assert len(temporary_allocations) == 1
+        (record,) = temporary_allocations
+        assert record.n_allocations == 10
```

Comment:
Hm. This is an implementation detail that depends on the initial size and may not be consistent across different C++ STL implementations, right?

Reply:
Yeah, but most are going to match it. We can go with this and change if we encounter something failing